doc/exts/pylint_messages.py

```diff
@@ -0,0 +1,207 @@
+    "F": "fatal",
+}
+
+
+class MessageData(NamedTuple):
+    checker: str
+    id: str
+    name: str
+    definition: MessageDefinition
+
+
+MessagesDict = Dict[str, List[MessageData]]
+OldMessagesDict = Dict[str, DefaultDict[Tuple[str, str], List[Tuple[str, str]]]]
+"""DefaultDict is indexed by tuples of (old name symbol, old name id) and values are
+tuples of (new name symbol, new name category)
+"""
+
+
+def _register_all_checkers_and_extensions(linter: PyLinter) -> None:
+    """Registers all checkers and extensions found in the default folders."""
+    initialize_checkers(linter)
+    initialize_extensions(linter)
+
+
+def _get_all_messages(
+    linter: PyLinter,
+) -> Tuple[MessagesDict, OldMessagesDict]:
+    """Get all messages registered to a linter and return a dictionary indexed by message
+    type.
+    Also return a dictionary of old message and the new messages they can be mapped to.
+    """
+    messages_dict: MessagesDict = {
+        "fatal": [],
+        "error": [],
+        "warning": [],
+        "convention": [],
+        "refactor": [],
+        "information": [],
+    }
+    old_messages: OldMessagesDict = {
+        "fatal": defaultdict(list),
+        "error": defaultdict(list),
+        "warning": defaultdict(list),
+        "convention": defaultdict(list),
+        "refactor": defaultdict(list),
+        "information": defaultdict(list),
+    }
+    for message in linter.msgs_store.messages:
+        message_data = MessageData(
+            message.checker_name, message.msgid, message.symbol, message
+        )
+        messages_dict[MSG_TYPES[message.msgid[0]]].append(message_data)
+
+        if message.old_names:
+            for old_name in message.old_names:
+                category = MSG_TYPES[old_name[0][0]]
+                old_messages[category][(old_name[1], old_name[0])].append(
+                    (message.symbol, MSG_TYPES[message.msgid[0]])
+                )
+
+    return messages_dict, old_messages
+
+
+def _write_message_page(messages_dict: MessagesDict) -> None:
+    """Create or overwrite the file for each message."""
+    for category, messages in messages_dict.items():
+        category_dir = os.path.join(PYLINT_BASE_PATH, "doc", "messages", category)
+        if not os.path.exists(category_dir):
+            os.makedirs(category_dir)
+        for message in messages:
+            messages_file = os.path.join(category_dir, f"{message.name}.rst")
+            with open(messages_file, "w", encoding="utf-8") as stream:
+                stream.write(f".. _{message.name}:\n\n")
+                stream.write(get_rst_title(f"{message.name} / {message.id}", "="))
+                stream.write("**Message emitted:**\n\n")
+                stream.write(f"{message.definition.msg}\n\n")
+                stream.write("**Description:**\n\n")
+                stream.write(f"*{message.definition.description}*\n\n")
+                stream.write(f"Created by ``{message.checker}`` checker\n")
```

Comment:
```suggestion
                title=get_rst_title(f"{message.name} / {message.id}", "=")
                stream.write(f""".. _{message.name}:

{title}    
**Message emitted:**

{message.definition.msg}

**Description:**

*{message.definition.description}*

                                                                                                                                                                                                                                Created by ``{message.checker}`` checker                                                                                                                                                                                                                                
"""
```
I think it's easier to see the formatting of a big text when using ``"""``

Reply:
Good idea!

Reply:
See https://github.com/PyCQA/pylint/pull/5396/commits/0b1e20b78f09c85fa87ee043050714693b242e9e